edge/pkg/devicetwin/dtcontroller.go

```diff
@@ -300,42 +300,32 @@ func classifyMsg(message *dttype.DTMessage) bool {
 		log.LOGGER.Infof("Classify the msg to action %s", action)
 		return true
 	} else if (strings.Compare(msgSource, "edgemgr") == 0) || (strings.Compare(msgSource, "devicecontroller") == 0) {
-		if strings.Contains(message.Msg.Router.Resource, "membership/detail") {
-			message.Action = dtcommon.MemDetailResult
+		switch message.Msg.Content.(type) {
```

Comment:
can you use an if else here?
With if else I think the readability would be increased.

Reply:
+1

Reply:
We could maybe use an assertion and if the assertion fails we can marshal the message ??

Reply:
I think it is better to keep the code style constant everywhere. We have a switch case in edged module

Reply:
fair enough